splitio/storage/pluggable.py

```diff
@@ -361,13 +448,12 @@ def get(self, segment_name):
         :rtype: splitio.models.segments.Segment
         """
         pass
-        # TODO: To be added when producer mode is aupported
-#        try:
-#            return segments.from_raw({'name': segment_name, 'added': list(self._pluggable_adapter.get(self._prefix.format(segment_name=segment_name))), 'removed': [], 'till': self._pluggable_adapter.get(self._segment_till_prefix.format(segment_name=segment_name))})
-#        except Exception:
-#            _LOGGER.error('Error getting segment')
-#            _LOGGER.debug('Error: ', exc_info=True)
-#            return None
+        try:
+            return segments.from_raw({'name': segment_name, 'added': list(self._pluggable_adapter.get(self._prefix.format(segment_name=segment_name))), 'removed': [], 'till': self._pluggable_adapter.get(self._segment_till_prefix.format(segment_name=segment_name))})
```

Comment:
Use adapter `get_items` method here (equivalent to `smembers` in redis.py).
Also remove `get_items_count` from SegmentsMockAdapter. It is not part of wrapper/adapter interface.

Reply:
in this line we are getting only one item, and we already use adapter.get() in SplitStorage, does it makes sense to add a duplicate call?

Reply:
I am referring to the code `'added': list(self._pluggable_adapter.get(self._prefix.format(segment_name=segment_name)))`. It should be `'added': self._pluggable_adapter.get_items(self._prefix.format(segment_name=segment_name))`

because `self._pluggable_adapter.get_items` is analog to `self._redis.smembers` in `RedisSegmentStorage::get`: it retrieves a list of items for the given key.

Reply:
fixed